File: solution/1300-1399/1359.Count All Valid Pickup and Delivery Options/Solution.java
class Solution {
    public int countOrders(int n) {
        final int mod = (int) 1e9 + 7;
        long f = 1;
        for (int i = 2; i <= n; ++i) {
            f = f * i * (2 * i - 1) % mod;
        }
        return (int) f;
    }
}

File: solution/1300-1399/1359.Count All Valid Pickup and Delivery Options/Solution.rs
const MOD: i64 = 1e9 as i64 + 7;

impl Solution {
    #[allow(dead_code)]
    pub fn count_orders(n: i32) -> i32 {
        let mut f = 1;
        for i in 2..=n as i64 {
            f = (i * (2 * i - 1) * f) % MOD;
        }
        f as i32
    }
}